packages/rainbowkit/src/components/ConnectButton/ConnectButton.tsx

```diff
@@ -42,248 +42,265 @@ export function ConnectButton({
   const chains = useRainbowKitChains();
   const connectionStatus = useConnectionStatus();
   const { setShowBalance } = useShowBalance();
+  const [ready, setReady] = useState(false);
 
   const { i18n } = useContext(I18nContext);
 
+  // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
   useEffect(() => {
     setShowBalance(showBalance);
+    if (!ready) setReady(true);
   }, [showBalance, setShowBalance]);
 
   return (
-    <ConnectButtonRenderer>
-      {({
-        account,
-        chain,
-        mounted,
-        openAccountModal,
-        openChainModal,
-        openConnectModal,
-      }) => {
-        const ready = mounted && connectionStatus !== 'loading';
-        const unsupportedChain = chain?.unsupported ?? false;
+    ready && (
```

Comment:
Only thing i did here was `ready && ...` didn't touch any other elements of the component